src/unistd.rs

```diff
@@ -438,37 +452,43 @@ pub fn dup(oldfd: RawFd) -> Result<RawFd> {
 /// specified fd instead of allocating a new one.  See the man pages for more
 /// detail on the exact behavior of this function.
 #[inline]
-pub fn dup2(oldfd: RawFd, newfd: RawFd) -> Result<RawFd> {
-    let res = unsafe { libc::dup2(oldfd, newfd) };
+pub fn dup2<Fd: AsFd>(oldfd: &Fd, newfd: &mut OwnedFd) -> Result<()> {
```

Comment:
Clever, taking `OwnedFd` by borrow.  But what if the user wants to supply an integer that doesn't correspond to any currently open file?  That was possible with the old interface.

Reply:
Dang, you're right. Will fix.

Reply:
> so I ended up adding dup*_raw variants.

This is something unfortunate, but I think this unfortunate approach is the only way to do it, rustix does not provide something similar to `unsafe fn dup2_raw()`, so one cannot specify a fd that is still not open.

And interestingly, rustix provides 3 helper functions to use `dup2` with `Stdin`, `Stdout` and `Stderr`: https://docs.rs/rustix/latest/rustix/stdio/fn.dup2_stdin.html

Reply:
> And interestingly, rustix provides 3 helper functions to use dup2 with Stdin, Stdout and Stderr: https://docs.rs/rustix/latest/rustix/stdio/fn.dup2_stdin.html

Now I think this is indeed necessary